framework/include/base/Assembly.h

```diff
@@ -1506,6 +1511,14 @@ class Assembly
   void
   computeSinglePointMapAD(const Elem * elem, const std::vector<Real> & qw, unsigned p, FEBase * fe);
 
+  /**
+   * Internal method for adding the cached residual from tag _residual_vector_tags[tag_index]._id
+   * into residual
+   *
+   * Note that this will also clear the cache
+   */
+  void addCachedResidualInternal(NumericVector<Number> & residual, unsigned int tag_index);
```

Comment:
`unsigned int` --> `TagID`

Reply:
But this isn't a `TagID`. It's an index into `_residual_vector_tags`, which contains only the tags that are residual tags. Therefore, if we have solution vector tags too, the indexing into `_residual_vector_tags` isn't the same as the ID of each tag in the vector.